File: clients/kotlin_graal/src/main/kotlin/ai_cup_22/model/Vec2.kt
package ai_cup_22.model

import ai_cup_22.util.StreamUtil

/**
 * 2 dimensional vector.
 */
class Vec2 {
    /**
     * `x` coordinate of the vector
     */
    var x: Double
    /**
     * `y` coordinate of the vector
     */
    var y: Double

    constructor(x: Double, y: Double) {
        this.x = x
        this.y = y
    }

    /**
     * Write Vec2 to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeDouble(stream, x)
        StreamUtil.writeDouble(stream, y)
    }

    /**
     * Get string representation of Vec2
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Vec2 { ")
        stringBuilder.append("x: ")
        stringBuilder.append(x)
        stringBuilder.append(", ")
        stringBuilder.append("y: ")
        stringBuilder.append(y)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Vec2 from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Vec2 {
            var x: Double
            x = StreamUtil.readDouble(stream)
            var y: Double
            y = StreamUtil.readDouble(stream)
            return Vec2(x, y)
        }
    }
}

File: clients/swift/Sources/AiCup22/Model/Vec2.swift
/// 2 dimensional vector.
public struct Vec2 {
    /// `x` coordinate of the vector
    let x: Double

    /// `y` coordinate of the vector
    let y: Double

    /// Read Vec2 from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Vec2 {
        var x: Double
        x = stream.readDouble()
        var y: Double
        y = stream.readDouble()
        return Vec2(x: x, y: y)
    }

    /// Write Vec2 to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeDouble(x)
        stream.writeDouble(y)
    }
}